packages/devextreme/js/__internal/ui/drawer/drawer.animation.ts

```diff
@@ -25,33 +26,26 @@ export interface FadeConfig {
 }
 
 export const animation = {
-  moveTo(config: DrawerMoveAnimationConfig): void {
-    const {
-      $element, position, direction = 'left', duration, complete,
-    } = config;
-    let toConfig = {} as AnimationState;
-    // eslint-disable-next-line no-undef-init
-    let animationType: AnimationType | undefined = undefined;
-
+  getMoveToConfig(direction: Direction, position: number): AnimationState | undefined {
     switch (direction) {
       case 'right':
-        // @ts-expect-error ts-error
-        toConfig = { transform: `translate(${position}px, 0px)` };
-        // @ts-expect-error ts-error
-        animationType = 'custom';
-        break;
+        return { transform: `translate(${position}px, 0px)` } as unknown as AnimationState;
       case 'left':
-        toConfig = { left: position };
-        animationType = 'slide';
-        break;
+        return { left: position };
       case 'top':
       case 'bottom':
-        toConfig = { top: position };
-        animationType = 'slide';
-        break;
+        return { top: position };
       default:
-        break;
+        return undefined;
     }
+  },
+
+  moveTo(config: DrawerMoveAnimationConfig): void {
+    const {
+      $element, position, direction = 'left', duration, complete,
+    } = config;
+    const toConfig = this.getMoveToConfig(direction, position);
+    const animationType: AnimationType = direction === 'right' ? ('custom' as unknown as AnimationType) : 'slide';
```

Comment:
```suggestion
    const animationType = direction === 'right' ? 'custom' : 'slide';
```

Reply:
It’s better to add @ts-expect-error before type: animationType, because it’s an issue in the public types. Once the accepted values for type are extended, the issue will be fixed